city_scrapers/spiders/il_corrections.py

```diff
@@ -0,0 +1,160 @@
+import re
+from datetime import datetime
+from io import BytesIO, StringIO
+
+import scrapy
+from city_scrapers_core.constants import ADVISORY_COMMITTEE
+from city_scrapers_core.items import Meeting
+from city_scrapers_core.spiders import CityScrapersSpider
+from pdfminer.high_level import extract_text_to_fp
+from pdfminer.layout import LAParams
+
+
+class IlCorrectionsSpider(CityScrapersSpider):
+    name = "il_corrections"
+    agency = "Illinois Department of Corrections Advisory Board"
+    timezone = "America/Chicago"
+    start_urls = [
+        "https://www2.illinois.gov/idoc/aboutus/advisoryboard/Pages/default.aspx"
+    ]
+
+    def __init__(self):
+        self.links = {}
+        self.url_base = "https://www2.illinois.gov"
+        self.pdf_text = ""
+
+    def parse(self, response):
+        """
+        `parse` should always `yield` Meeting items.
+
+        Change the `_parse_title`, `_parse_start`, etc methods to fit your scraping
+        needs.
+        """
+        # Gather Meeting Dates/Links
+        self.links = self._parse_all_links(response)
+
+        for date in self.links.keys():
+            if "Agenda" in self.links[date].keys():
+                yield scrapy.Request(
+                    self.links[date]["Agenda"],
+                    callback=self._meeting,
+                    cb_kwargs=dict(date=date),
+                )
+            elif "Minutes" in self.links[date].keys():
+                yield scrapy.Request(
+                    self.links[date]["Minutes"],
+                    callback=self._meeting,
+                    cb_kwargs=dict(date=date),
+                )
+
+    def _meeting(self, response, date):
+        self.pdf_text = self._parse_pdf(response)
```

Comment:
Mentioned above, but it could be more clear to set this as `pdf_text` rather than `self.pdf_text` and pass it as an argument to the functions where it's needed so that this is a little more clear